sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -81,26 +82,27 @@
     /*
      * This will save last poll response.
      */
-    private PollResponse<T> pollResponse;
+    private final AtomicReference<PollResponse<T>> pollResponse = new AtomicReference<>(
```

Comment:
I am not sure if using AtomicReference here is adding much value as the code below doesn't really ensure atomicity.

Reply:
Removed and reverted back to volatile.